extension.js

```diff
@@ -354,6 +360,17 @@ function update_height_setting(win) {
     const workarea = workarea_for_window(win);
     const current_height = win.get_frame_rect().height / workarea.height;
     settings.set_double('window-height', Math.min(1.0, current_height));
+    settings.set_boolean('window-maximize', false);
+}
+
+function window_maximize() {
+    if (!current_window)
+        return;
+
+    if (current_window.maximized_vertically)
```

Comment:
If `window_maximize` is called in response to `window-maximize` changes, you should read `window-maximize` in it. Otherwise it's not obvious that the value can't become out of sync.

Also, if the function also does `unmaximize`, it shouldn't be called `window_maximize`. Maybe `set_[window_]maximized`, `update_maximized`, `sync_maximized`, etc. (Yes, I know, there are already multiple poorly named functions, let's not add another one)

Reply:
you're right, I'll do "toggle" since it is the name of the option as well.

Reply:
From a `toggle_*()` function I expect that it will invert some boolean (if it was enabled, disable; if it was disabled - enable). Though I'm not an expert in English and may be wrong :) I called similar functions simply `set_*()`: see [`set_window_above`](https://github.com/amezin/gnome-shell-extension-ddterm/blob/master/extension.js#L252), `set_window_stick` - they also set (update) window properties based on boolean settings

Reply:
not my first language either so who am I to say :P I've changed to `set`. It sounds good to me.